### PR TITLE
Fix AB-293: Proper exit status for files

### PR DESCRIPTION
**Pull request by rsh7**
I am first reading the reason column from the `filelog` and printing this reason before moving on to the next file.

### CODE REVIEW

**Comment by alastair** [comment]
Thanks for your contribution. Unfortunately we can't accept this patch as it currently is.
We want the number of dependencies of this software to be as low as possible. In fact, we include `requests` in the package and don't require any other external software. For this reason we don't want to use pandas - it would require our users to install this software separately in order to be able to use the tool.

Having said that, there is no need for pandas here - we should be using the standard sqlite library to access the file log.

The `_update_progress` method is a utility used only for printing messages to the screen. There should be no logic in this method. In fact, I can't even see how you get the `filepath` variable in this method.

As I mentioned in the previous PR, we should read the status from the database. A new method `get_processed_status` could return `GREEN, ":) done"` if the file was previously successfully processed (that is, the reason column in the filelog table is null), or `RED, ":( nombid"` if there is a reason in the column.
It would also have to return something based on the `extractor` reason. Note that the printed message in this case just says `:( extract` - this is so that it fits within the `[  ]` brackets, which only have . You'll also have to return the colour to pass to `_update_status`

**Comment by RashiSah** [comment]
On reprocessing files, `get_processed_method` is reading the value of `filelog.reason` for the particular `filepath` from database and returning the status based on the values.

File: abz/acousticbrainz.py [hunk 1]
```diff
@@ -67,6 +67,12 @@ def is_processed(filepath):
     else:
         return False
 
+def get_processed_status(filepath):
```
Comment (Freso) [inline abz/acousticbrainz.py]: There should be an additional blank line above here: https://www.python.org/dev/peps/pep-0008/#blank-lines

File: abz/acousticbrainz.py [hunk 2]
```diff
@@ -67,6 +67,12 @@ def is_processed(filepath):
...
+    query = """select reason from filelog where filename = ?"""
+    c = conn.cursor()
+    c.execute(query,(filepath,))
+    processed_status = c.fetchone()
+    return processed_status[0]
```
Comment (Freso) [inline abz/acousticbrainz.py]: There should be an additional blank line below here: https://www.python.org/dev/peps/pep-0008/#blank-lines

File: abz/acousticbrainz.py [hunk 3]
```diff
@@ -91,7 +97,14 @@ def submit_features(recordingid, features):
 def process_file(filepath):
     _start_progress(filepath)
     if is_processed(filepath):
-        _update_progress(filepath, ":) done", GREEN)
+        processed_status = get_processed_status(filepath)
+        if processed_status == None:
+	        status = ":) done"
+	        colour = GREEN
+        else:
+	        status = ":( nombid"
```
Comment (Freso) [inline abz/acousticbrainz.py]: There are other reasons for failing than "nombid", as @alastair also mentions in his comment. I get this when I run `SELECT reason FROM "main"."filelog" WHERE reason <> 'NULL';` on my local abzsubmit db:
```
"nombid"
"nombid"
"nombid"
"nombid"
"nombid"
"nombid"
"nombid"
"nombid"
"nombid"
"nombid"
"nombid"
"nombid"
"nombid"
"nombid"
"nombid"
"extractor"
"extractor"
"extractor"
"nombid"
"nombid"
```

There might be other errors too.

File: abz/acousticbrainz.py [hunk 4]
```diff
@@ -91,7 +97,14 @@ def submit_features(recordingid, features):
 def process_file(filepath):
     _start_progress(filepath)
     if is_processed(filepath):
-        _update_progress(filepath, ":) done", GREEN)
+        processed_status = get_processed_status(filepath)
+        if processed_status == None:
+	        status = ":) done"
```
Comment (Freso) [inline abz/acousticbrainz.py]: You're using more than 4-space indentation for this last level, while 4-space indentation levels are used for the rest of the function definition. Also: https://www.python.org/dev/peps/pep-0008/#indentation

File: abz/acousticbrainz.py (hunk 1 above)
Comment (RashiSah) [reply abz/acousticbrainz.py]: @Freso I have added a blank line here :https://github.com/MTG/acousticbrainz-client/pull/46/commits/a252727f5056946f097c1a94b52a323658a6aa18

File: abz/acousticbrainz.py (hunk 2 above)
Comment (RashiSah) [reply abz/acousticbrainz.py]: Done!

File: abz/acousticbrainz.py (hunk 4 above)
Comment (RashiSah) [reply abz/acousticbrainz.py]: Now the entire function definition follows 4-space indentation :)

**Comment by RashiSah** [comment]
- This is a
    - [x] Bug Fix
    - [ ] Feature addition
    - [ ] Refactoring
    - [ ] Minor/simple change
    - [ ] Other

**Problem Statement**
If we submit audio files for the first time which has no MBID, the exit status comes out to be `[:( nombid ]`
After that any number of times if we submit the same files and without any MBID, the exit status changes to `[:) done ]`, which is not expected.

**Solution**
Exit status should be same on repeated trials of submissions of same set of files. Now, a method `get_status ` is getting the status from the filelog table by reading the reason column for every music file one by one. Now, the colour and status which are getting passed to `_update_status`, are `GREEN, ":) done"` if the file was previously successfully processed, `RED, ":( extract"` for extractor reason and `RED,  ":( nombid"` if there is any other reason in the column.
Now, the proper exit status will be displayed for every acousticbrainz-client submissions of music files.

**Comment by alastair** [comment]
Thanks for the patch! I've merged your changes but also made some other updates. You can see them in https://github.com/MTG/acousticbrainz-client/commit/92078ecd583e9103bd8b83e02e03a7a9fc08d479

Instead of a separate `is_processed` and `get_processed_status` method, I made a single method `get_processed_status` which checks **and** returns the value. As @Freso noted, there are many types of error status. Your updated change added `nombid`, but there is also `json`, added here: https://github.com/MTG/acousticbrainz-client/blob/92078ecd583e9103bd8b83e02e03a7a9fc08d479/abz/acousticbrainz.py#L151
By getting the status directly from the database we don't need to have the if/elsif block for all cases. If we add another message in the future we won't have to add another check.

**Pull request merged**